core/drivers/crypto/stm32/stm32_cryp.c

```diff
@@ -1277,6 +1277,10 @@ static TEE_Result stm32_cryp_driver_init(void)
 	if (stm32_reset_deassert(cryp_pdata.reset_id, TIMEOUT_US_1MS))
 		panic();
 
+	res = stm32_register_authenc();
```

Comment:
To support `CFG_CRYPTO_DRV_AUTHENC` and `CFG_CRYPTO_DRV_CIPHER` being exclusively enabled, prefer:
```c
	if (IS_ENABLED(CFG_CRYPTO_DRV_AUTHENC)) {
		res = stm32_register_authenc();
		if (res)
			return res;
	}

	if (IS_ENABLED(CFG_CRYPTO_DRV_CIPHER)) {
		res = stm32_register_cipher();
		if (res)
			return res;
	}

	return TEE_SUCCESS;
}
```

By the way, if one enables these CFG_* and use a compliant DT node, I think here we should panic since we expect CRYP support and should not find a fallback (i.e SW implementation). All in one.
```patch
 	if (IS_ENABLED(CFG_CRYPTO_DRV_AUTHENC)) {
 		res = stm32_register_authenc();
 		if (res) {
-			return res;
+			EMSG("Failed to register to authenc: %#"PRIx32, res);
+			panic();
		}
	}

Reply:
Shouldn't do the the same in commit "crypto: stm32: use CRYP IP for CIPHER algorithms" ?
ie panic() if register_cipher fail ?

Reply:
yes, I think so.